### DISCUSSION TITLE
How to document docker containers in Netbox?

### DISCUSSION THREAD

**Discussion by XGhosted** [netbox-community/netbox#9655] [Q&A] [answered]
Hi, I was wondering how best to document docker containers in Netbox. Right now I have about 20+ containers running but only documented in a self hosted wiki.

I would like to have this documented in Netbox and my question was how do others document containers in Netbox?

**Comment by candlerb** [comment, marked as answer]
If these are long-running docker containers, I would document them as Virtual Machines.

If you are using "docker run" or "podman run" on standalone hosts, then create separate "clusters" each containing a single host.  Create a cluster type of "docker".  Then you can associate each container with the correct "cluster" host.

If it's a kubernetes cluster (or docker swarm or nomad or whatever), where containers are potentially mobile between a group of hosts, then make a single "cluster" containing the group of hosts.  Again, you can create a cluster type of "kubernetes", "swarm" etc.

If you run your docker containers inside a virtual machine, then I'd still create a "cluster" to represent the containers running inside that VM. You can't directly associate the docker cluster with a VM instance (as it's not a device), but you can add a comment or a custom field.  A cluster with zero devices is still legitimate.

**Reply by nklamann** [reply]
> I would document them as Virtual Machines.

This is only possible if the container does not run on a virtual machine by itself.
